packages/itwin/tree-widget/src/components/trees/common/UseVisibilityTree.tsx

```diff
@@ -0,0 +1,65 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+
+import { useCallback, useEffect } from "react";
+import { VisibilityTreeEventHandler } from "../VisibilityTreeEventHandler";
+import { usePresentationTreeState } from "@itwin/presentation-components";
+
+import type { IModelConnection } from "@itwin/core-frontend";
+import type { Ruleset } from "@itwin/presentation-common";
+import type { IFilteredPresentationTreeDataProvider, PresentationTreeEventHandlerProps, UsePresentationTreeStateProps } from "@itwin/presentation-components";
+import type { IVisibilityHandler, VisibilityTreeSelectionPredicate } from "../VisibilityTreeEventHandler";
+import type { VisibilityTreeFilterInfo } from "./Types";
+
+/**
+ * Props for [[useVisibilityTree]] hook.
+ * @beta
+ */
+export interface UseVisibilityTreeProps extends Omit<UsePresentationTreeStateProps<VisibilityTreeEventHandler>, "rulesetOrId"> {
+  imodel: IModelConnection;
+  ruleset: Ruleset;
+  visibilityHandler?: IVisibilityHandler;
+  filterInfo?: VisibilityTreeFilterInfo;
+  onFilterChange?: (filteredDataProvider?: IFilteredPresentationTreeDataProvider, matchesCount?: number) => void;
+  selectionPredicate?: VisibilityTreeSelectionPredicate;
+}
+
+/**
+ * Custom hooks that creates state for visibility tree using [[usePresentationTreeState]] hook.
```

Comment:
The `[[usePresentationTreeState]]` won't work since it's in another package. I don't think we even need to mention it here.

```suggestion
 * Custom hook for creating visibility tree component state.
```